server/controllers/profilephoto.js

```diff
@@ -84,13 +80,8 @@ exports.downloadPhoto = asyncHandler(async (req, res) => {
                 if (error) {
                     res.json("Failed to retrieve an object: ");
                 } else {
-                    fs.writeFile("../client/src/Images/" + profile.photoUrl.substring(72), data.Body, (err) => {
-                        if (err)
-                            res.json('Error occured while writing file');
-                        else {
-                            res.json("File written successfully");
-                        }
-                    });
+                    res.setHeader("content-type", "some/type");
+                    fs.createReadStream(`./${profile.photoUrl.substring(72)}`).pipe(res);

```

Comment:
Have you tried this to see if it works? I don't see where we use the `data` we download from s3 here 🤔